### PR TITLE
Rorys hpstr

### PR DESCRIPTION
**Pull request by rodwyer100**
Pushing in the desired changes from Cam into the current branch. Made rawSvtAna_cfg.py changes to reflect the names of new regions I am using. I may include new config files should Cam think it appropriate for this commit. The name removal thing (to optionally not include hits in your fsp) is as of yet invalidated. Will check now.

### CODE REVIEW

File: processors/config/recoTuple_cfg.py [hunk 1]
```diff
@@ -146,9 +147,14 @@
 fsp.parameters["fspCollRoot"] = "FinalStateParticles_KF"
 fsp.parameters["kinkRelCollLcio"] = ""
 fsp.parameters["trkRelCollLcio"] = "KFTrackDataRelations"
+
 if(rawHits==1):
```
Comment (cbravo135) [inline processors/config/recoTuple_cfg.py]: This needs to be options.rawHits

Comment (rodwyer100) [reply processors/config/recoTuple_cfg.py]: Fixed the two places it occured.

**Comment by rodwyer100** [comment]
Validation Task:

I first run a default fsp processor on a collection of reconstructed events from 14552 (I already run data_cnv jobs on them for another task) and obtain the following output with options.rawHits=1:
image

When I run with -r = 0 or options.rawHits=0 I obtain the following output:
image

This is consistent with what we desired: it does remove the names of collections in fsp relevant to keeping hits

This produced two output root files, one clearly missing the Hit collections:
image

So far it is passing validation testing with flying colors. One thing in TBrowser is that when I click on distributions in FSP which should be populated they are not plotted on TBrowser. There are alot of reasons this can occur which should not cause any concern, but I will fish around a little longer.

**Comment by rodwyer100** [comment]
Because I am a world class dingus, I forgot this was applied to slcio files :). Runnign on them with the same theme (also including header in fsp) I get the following:
image

Even in the collection without hits, track information is indeed now populated:
image

Everything seems kosher now.

**Comment by bloodyyugo** [comment]
I assume this works the same if we include the full list of Si clusters and raw hits instead of just the hits-on-track.  True?

**Comment by rodwyer100** [comment]
Sorry Matt let me reply to Cam first: he asked me to do something beefy and it took me awhile. Namely he wanted me to validate that when I remove raw htis and keep the hits on track I can still acess the hits on track. I can. Here are their times:
image

**Comment by rodwyer100** [comment]
Sorry Matt maybe I just answered your question as well? It should if I understand your meaning correctly. Thats just including all the relevant info. You can the raw hits and other stuff for analysis optionally

**Comment by rodwyer100** [comment]
I think you would need to include track hits in any case because you need them to refer to raw hits (as they are contained as a property of them and are linked by a TRefArray).

**Comment by cbravo135** [comment]
@bloodyyugo The Ntuples are configurable to keep what you want. If you want to include all the raw hits, there is a different processor that already exists that you should use, and same with Si clusters. This processor will include the hits and/or raw hits (configurable) associated to the particle tracks.

File: processors/config/fspTuple_cfg.py [hunk 2]
```diff
@@ -55,12 +55,15 @@
...
+    fsp.parameters["trkhitCollRoot"] = "fspOnTrackHits"
+    fsp.parameters["rawhitCollRoot"] = "fspOnTrackRawHits"
     fsp.parameters["hitFitsCollLcio"] = "SVTFittedRawTrackerHits"
 else:
+    fsp.parameters["trkhitCollRoot"] = "fspOnTrackHits"
+    fsp.parameters["rawhitCollRoot"] = ""
     fsp.parameters["hitFitsCollLcio"] = ""
-
-sequence = []
+sequence = [header]
```
Comment (cbravo135) [inline processors/config/fspTuple_cfg.py]: The code would be cleaner if you just did p.sequence = [header, fsp]
Three lines could be achieved with one.

Comment (rodwyer100) [reply processors/config/fspTuple_cfg.py]: Fixed it :)

**Comment by rodwyer100** [comment]
Added all of my config files for rawSvtHitAnaProcessor. Should be able to run now.

**Review by cbravo135 — APPROVED** [review]

**Pull request merged**